src/main/java/com/puppycrawl/tools/checkstyle/api/TokenTypes.java

```diff
@@ -6469,6 +6469,244 @@ public final class TokenTypes {
+
+    /**
+     * The opening delimiter of an embedded expression within a string template.
+     * <p>For example:</p>
+     * <pre>
+     *     String s = STR."Hello, \{name}!";
+     * </pre>
+     * <p>parses as:</p>
+     * <pre>
+     * VARIABLE_DEF -&gt; VARIABLE_DEF
+     *  |--MODIFIERS -&gt; MODIFIERS
+     *  |--TYPE -&gt; TYPE
+     *  |   `--IDENT -&gt; String
+     *  |--IDENT -&gt; s
+     *  `--ASSIGN -&gt; =
+     *      `--EXPR -&gt; EXPR
+     *           `--DOT -&gt; .
+     *               |--IDENT -&gt; STR
+     *               `--STRING_TEMPLATE_BEGIN -&gt; "
+     *                   |--STRING_TEMPLATE_CONTENT -&gt; Hello,
+     *                   |--EMBEDDED_EXPRESSION_BEGIN -&gt; \{
+     *                   |--EMBEDDED_EXPRESSION -&gt; EMBEDDED_EXPRESSION
+     *                   |   `--IDENT -&gt; name
+     *                   |--EMBEDDED_EXPRESSION_END -&gt; }
+     *                   |--STRING_TEMPLATE_CONTENT -&gt; !
+     *                   `--STRING_TEMPLATE_END -&gt; "
+     * </pre>
+     *
+     * @see #STRING_TEMPLATE_END
+     * @see #STRING_TEMPLATE_CONTENT
+     * @see #EMBEDDED_EXPRESSION_BEGIN
+     * @see #EMBEDDED_EXPRESSION
+     * @see #EMBEDDED_EXPRESSION_END
+     * @see #STRING_LITERAL
+     *
+     * @since 10.13.0
+     */
+    public static final int EMBEDDED_EXPRESSION_BEGIN =
+            JavaLanguageLexer.EMBEDDED_EXPRESSION_BEGIN;
+
+    /**
+     * An expression embedded within a string template.
+     * <p>For example:</p>
+     * <pre>
+     *     String s = STR."Hello, \{name}!";
+     * </pre>
+     * <p>parses as:</p>
+     * <pre>
+     * VARIABLE_DEF -&gt; VARIABLE_DEF
+     *  |--MODIFIERS -&gt; MODIFIERS
+     *  |--TYPE -&gt; TYPE
+     *  |   `--IDENT -&gt; String
+     *  |--IDENT -&gt; s
+     *  `--ASSIGN -&gt; =
+     *      `--EXPR -&gt; EXPR
+     *           `--DOT -&gt; .
+     *               |--IDENT -&gt; STR
+     *               `--STRING_TEMPLATE_BEGIN -&gt; "
+     *                   |--STRING_TEMPLATE_CONTENT -&gt; Hello,
+     *                   |--EMBEDDED_EXPRESSION_BEGIN -&gt; \{
+     *                   |--EMBEDDED_EXPRESSION -&gt; EMBEDDED_EXPRESSION
+     *                   |   `--IDENT -&gt; name
+     *                   |--EMBEDDED_EXPRESSION_END -&gt; }
+     *                   |--STRING_TEMPLATE_CONTENT -&gt; !
+     *                   `--STRING_TEMPLATE_END -&gt; "
+     * </pre>
+     *
+     * @see #STRING_TEMPLATE_END
+     * @see #STRING_TEMPLATE_CONTENT
+     * @see #EMBEDDED_EXPRESSION_BEGIN
+     * @see #EMBEDDED_EXPRESSION
+     * @see #EMBEDDED_EXPRESSION_END
+     * @see #STRING_LITERAL
+     *
+     * @since 10.13.0
+     */
+    public static final int EMBEDDED_EXPRESSION =
+            JavaLanguageLexer.EMBEDDED_EXPRESSION;
+
+    /**
+     * The closing delimiter of an embedded expression within a string
+     * template.
+     * <p>For example:</p>
+     * <pre>
+     *     String s = STR."Hello, \{name}!";
```

Comment:
`String s = STR."Hello, \{getName("Mr. ")}!";`

Reply:
Changed this one a bit to reuse the firstName, lastName concept but with a method call as suggested.